docs/madr/decisions/046-meshservice-policy-matching.md

```diff
@@ -0,0 +1,322 @@
+
+Top level zone would look like the following:
+
+```yaml
+kind: NonLocalMeshService
+namespace: other-zone-ns
+zone: other-zone
+name: name-in-its-zone
+```
+
+Using `zone` top level could allow us to be "smarter" about selecting
+cross-zone `MeshServices`. Instead of matching the actual Kubernetes object
+we could transparently match `name`/`namespace` to `kuma.io/display-name`/`kuma.io/namespace`, instead of
+requiring:
+
+```yaml
+kind: NonLocalMeshService
+labels:
+  kuma.io/zone: other-zone
+  kuma.io/display-name: name-in-its-zone
+  k8s.kuma.io/namespace: other-zone-ns
+```
+
+which is significantly more noisy. Using `labels` is less "magic" and doesn't
+look like we're matching an object that doesn't actually exist in the cluster.
+
+##### Positive
+
+Allows us to be explicit as well as not back ourselves into a corner
+with the API.
+
+##### Negative
+
+* more verbose
+* not always directly matching Kubernetes objects
+
+#### `sectionName`
+
+##### Positive
+
+* consistency with Gateway API
+
+##### Negative
+
+* too generic, potentially confusing
+
+#### `NonLocalMeshService`
+
+```yaml
+kind: NonLocalMeshService
+namespace: other-zone-ns
+zone: other-zone
+```
+
+##### Positive
+
+* sidesteps issues of the k8s namespace not being the same as the original k8s
+  namespace
+
+##### Negative
+
+* one more additional kind
+
+#### `MeshServiceId`
+
+```yaml
+kind: MeshServiceId
+name: kuma://backend/zone-1/backend-ns/8080
+```
+
+##### Positive
+
+* least verbose of the options
+
+##### Negative
+
+* one more additional kind
+* not structured
+
+## Decision Outcome
+
+We choose:
+
+* use generic Gateway API targetRef field `sectionName` to specify `port` and
+  add `labels` but use `port` in `backendRefs`.
+* use exactly one of `labels` or `name`/`namespace`
+* no new top level fields besides `labels`
+
+### `targetRef`
+
+The policy is applied
+on both the Envoy outbound listeners and clusters
+that correspond to the `MeshService`.
+Note that more than one `MeshService` can be matched in which case it applies to
+all matched `MeshServices`.
+
+References for `kind: MeshService` have the following structure:
+
+```yaml
+---
+kind: MeshService
+spec:
+  ports:
+  - port: 80
+    name: http
+---
+targetRef:
+  kind: MeshService
+  namespace: backend
+  name: backend # only name/namespace or labels
+  labels: # only name/namespace or labels
+    kuma.io/zone: east
+  sectionName: http
+```
+
+The `name` of a `MeshService` ref always refers to the name of one, real Kubernetes
+object in the zone of the policy.
+
+Exactly one of either `targetRef.name` or `targetRef.labels` is **required**.
+
+#### `sectionName`
+
+If `sectionName` is set in `targetRef.kind` with `kind: MeshService`,
+it refers to an entry in `ports` by name and only traffic to that port is affected.
+
+In `backendRefs`, `port` is required. There is no `sectionName`, in accordance
+with Gateway API HTTPRoute.
+
+#### `name`
```

Comment:
The backwards compatibility is not covered, but I think we discussed that 
```yaml
to:
  - targetRef:
      kind: MeshService
      name: backend
```
should try to find the real MeshService resource named `backend` first and if there is none then work as an old-fashioned MeshService?